packages/core/tests/workflow.test.ts

```diff
@@ -113,15 +113,16 @@ describe("Workflow", () => {
     );
   });
 
-  test("workflow validation", async () => {
-    const jokeFlow = new Workflow({ verbose: true, validate: true });
-    jokeFlow.addStep(StartEvent, generateJoke, { outputs: StopEvent });
-    jokeFlow.addStep(JokeEvent, critiqueJoke, { outputs: StopEvent });
-    const run = jokeFlow.run("pirates");
-    await expect(run).rejects.toThrow(
-      "The following events are consumed but never produced: JokeEvent",
-    );
-  });
+  // fixme: design a better API for validation
+  // test("workflow validation", async () => {
```

Comment:
This is seems like a helper function for in/out Event relationship.

But should we check this in runtime, instead of top of the calling run